### PR TITLE
Gemini -(Kumo) Task created in a session that has been invalidated

### PR DESCRIPTION
**Pull request by gnkeshavamurthy**
Added method to update session headers instead of recreating new session everytime

### CODE REVIEW

File: Sources/Kumo/Services/Service.swift [hunk 1]
```diff
@@ -145,6 +145,13 @@ public class Service {
 
     internal func copySettings(from _: ApplicationLayer) { }
 
+    /// Update the header with value for HTTP.Header type
+    /// - Parameters:
+    ///   - value: any vlaue for the HTTP.Header
```
Comment (powerje) [inline Sources/Kumo/Services/Service.swift:150]:
```suggestion
    ///   - value: any value for the HTTP.Header
```

**Review by powerje — APPROVED** [review]
Approved after fixing that typo

Makes me wonder why we didn't use this approach before 🤔 Should we deprecate the `reconfiguring` options as they seem error prone?

**Comment by rpwachowski** [comment]
> Approved after fixing that typo
> 
> 
> 
> Makes me wonder why we didn't use this approach before 🤔 Should we deprecate the `reconfiguring` options as they seem error prone?

https://developer.apple.com/documentation/foundation/urlsession/1411477-configuration

The documentation reports that changes are not respected; are they actually being respected?

**Comment by rpwachowski** [comment]
The property returns a copy of its own configuration.

**Comment by powerje** [comment]
> > Approved after fixing that typo
> > Makes me wonder why we didn't use this approach before 🤔 Should we deprecate the `reconfiguring` options as they seem error prone?
> 
> https://developer.apple.com/documentation/foundation/urlsession/1411477-configuration
> 
> The documentation reports that changes are not respected; are they actually being respected?

Oof, I looked through the docs to find that this morning when I approved it but totally missed it. This would explain why we didn't go this route before.

@gnkeshavamurthy is the property update actually being respected here?

cc: @pavm035

**Review by powerje — CHANGES_REQUESTED** [review]
Approval removed for now while checking into whether or not that configuration change is actually used (docs seem clear that they will not be)

**Pull request closed without merging**